tests/utils/test_requirement_checker.py

```diff
@@ -171,10 +172,60 @@ def test_install_only_one_alternative(self, behavior_requirements_file):
     def test_does_not_install_if_alternative_is_present(
         self, behavior_requirements_file
     ):
+        with mock.patch(
+            "subprocess.check_output", wraps=replace_subprocess_check_output
+        ):
+            # doesn't install if one of the alternative is available
+            assert check_requirements(NODE_WITH_UPDATE, behavior_requirements_file) == 3
+
+    @pytest.mark.parametrize(
+        "behavior_requirements_file", ("python + selector",), indirect=True
+    )
+    def test_skip_install_without_selector(self, behavior_requirements_file):
         with mock.patch(
             "subprocess.check_output", wraps=replace_subprocess_check_output
         ), TestCase.assertLogs(
             "peekingduck.utils.requirement_checker.logger"
         ) as captured:
-            # doesn't install if one of the alternative is available
-            assert check_requirements(NODE_WITH_UPDATE, behavior_requirements_file) == 3
+            # does not install any req with selector, so 3 updates
+            n_update = check_requirements(
+                NODE_WITH_UPDATE, behavior_requirements_file, None
+            )
+
+            assert n_update == 3
+            assert_msg_in_logs(f"{''.join(PY_PKGS[2])} not found", captured.records)
+            assert_msg_in_logs(f"pip install {''.join(PY_PKGS[2])}", captured.records)
+
+    @pytest.mark.parametrize(
+        "behavior_requirements_file", ("python + selector",), indirect=True
+    )
+    @pytest.mark.parametrize("flags", ("flag2", "invalid_flag"))
```

Comment:
Can `flag2` be named into something meaningful like `invalid_flag`? Maybe `valid_flag`?

Reply:
Changed